### PR TITLE
Fix various issues related to building RPMs

### PR DESCRIPTION
**Pull request by brettle**
Some of these commits are the same as what was included in the recent PRs from @marvelous, but I've merged with your latest master so there shouldn't be any conflicts.

With these changes and [the enki PR I recently submitted](https://github.com/enki-community/enki/pull/9), RPMs are successfully building on Fedora 20, 21, and 22 and OpenSUSE 13.1 and 13.2. Repositories are available [here](https://build.opensuse.org/project/repositories/home:brettle:aseba).

### CODE REVIEW

File: menu/freedesktop/asebaonlinehelp.desktop [hunk 1]
```diff
@@ -7,10 +7,4 @@ Name[es]=Ayuda en línea Aseba
...
 URL=http://www.thymio.org/en:start
-URL[fr]=http://www.thymio.org/fr:start
-URL[de]=http://www.thymio.org/de:start
-URL[es]=http://www.thymio.org/es:start
-URL[it]=http://www.thymio.org/it:start
-URL[pt]=http://www.thymio.org/en:start
-Categories=Education;Science;Aseba
-Icon=asebahelp
\ No newline at end of file
+Icon=asebahelp
```
Comment (stephanemagnenat) [inline menu/freedesktop/asebaonlinehelp.desktop]: Why these changes? The URL should depend on the language. Is it unvalid?

On Kubuntu, the URL did not appear in the menu, is that the reason?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -195,6 +195,8 @@ if (LIBXML2_FOUND)
         include_directories(${LIBXML2_INCLUDE_DIR})
 	include_directories(externals/Catch/include ${COMMON_INCLUDES})
 	add_executable(test-asebahttp test-http.cpp)
+	# Use set_target_properties instead of target_compile_options to make CMake <= 2.8.11 happy
+	set_target_properties(test-asebahttp PROPERTIES COMPILE_FLAGS -std=c++11)
```
Comment (stephanemagnenat) [inline tests/CMakeLists.txt]: Why do we need c++11 support here?

**Comment by stephanemagnenat** [comment]
There are some whitespace changes. Do you think you can remove them to have cleaner patches?

Also, do you think it would make sense to rebase the pull request on master to have less commits?

File: menu/freedesktop/asebaonlinehelp.desktop (hunk 1 above)
Comment (brettle) [inline menu/freedesktop/asebaonlinehelp.desktop]: A desktop file linter pointed out that URL entries are not localizable. [The spec says](http://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys) that URL entries are of type `string`, not `localestring`.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (brettle) [inline tests/CMakeLists.txt]: I ran into philsquared/Catch#424 when building on Fedora 22. Note that the problem isn't with my particular machine as is suggested in that issue. I saw the problem on the build farm too. I suspect it is a GCC 5 bug.

File: menu/freedesktop/asebaonlinehelp.desktop (hunk 1 above)
Comment (stephanemagnenat) [inline menu/freedesktop/asebaonlinehelp.desktop]: Ok, I missed that one in the doc. Thanks for correcting it.

Btw, why did you remove Categories for this entry?

File: tests/CMakeLists.txt (hunk 2 above)
Comment (stephanemagnenat) [inline tests/CMakeLists.txt]: Ok, so maybe the comment above the set_target_properties line should be updated? It does not seem very informative about why we have this line.

File: menu/freedesktop/asebaonlinehelp.desktop (hunk 1 above)
Comment (brettle) [inline menu/freedesktop/asebaonlinehelp.desktop]: Linter complained about that too. The same section of the spec says that "Categories" is only allowed on Application (i.e. type 1) entries, not Link (i.e. type 3) entries.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (brettle) [inline tests/CMakeLists.txt]: Yeah. Also, it looks like [there is a way](https://github.com/philsquared/Catch/blob/master/docs/configuration.md#c-conformance-toggles) to tell Catch that the compiler doesn't support C++11. I'm thinking it's better to do that so that aseba is buildable by more compilers. If you disagree, let me know. Otherwise I'll try making that change later tonight.

File: menu/freedesktop/asebachallenge.desktop [hunk 3]
```diff
@@ -9,5 +9,5 @@ GenericName[es]=Aseba simulador de robot para jugar en la red
 GenericName[pt]=Aseba simulador de robô para jogar em rede
 Exec=asebachallenge
 TryExec=asebachallenge
-Categories=Education;Science;Aseba
+Categories=Education;Science;Aseba;
```
Comment (stephanemagnenat) [inline menu/freedesktop/asebachallenge.desktop]: There is an added ";", should it also be in all other .desktop files?

Comment (brettle) [inline menu/freedesktop/asebachallenge.desktop]: Yes, but it's not critical. During the package build process I just get warnings like this:

```
[  346s] + desktop-file-install --remove-category=Aseba --dir=/home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebachallenge.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebaonlinehelp.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebaplayground.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebastudio.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebastudiothymio.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/thymioupgrader.desktop /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/thymiovpl.desktop
[  346s] /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebaplayground.desktop: warning: key "Categories" is a list and does not have a semicolon as trailing character, fixing
[  346s] /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebastudio.desktop: warning: key "Categories" is a list and does not have a semicolon as trailing character, fixing
[  346s] /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/asebastudiothymio.desktop: warning: key "Categories" is a list and does not have a semicolon as trailing character, fixing
[  346s] /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/thymioupgrader.desktop: warning: key "Categories" is a list and does not have a semicolon as trailing character, fixing
[  346s] /home/abuild/rpmbuild/BUILDROOT/aseba-1.4.0-0.2.20150912git04b42d4.fc22.x86_64/usr/share/applications/thymiovpl.desktop: warning: key "Categories" is a list and does not have a semicolon as trailing character, fixing
```
If you want me to do another round of edits to the desktop files, I will. But if I do, I'm thinking we should also remove the "Aseba" category itself. I'm not sure why it's there, but the spec file runs desktop-file-install with --remove-category=Aseba so it's being removed during RPM packaging. Also, if you run desktop-file-validate on the desktop files, it complains that Aseba is not a standard category.

So, should I edit all the desktop files' Categories to remove "Aseba" and add a trailing ";" if necessary? Or do you just want to stick with it as is since the packages all built successfully)?

Comment (stephanemagnenat) [inline menu/freedesktop/asebachallenge.desktop]: If I am using something non-standard, let's correct it! The reason I used the Aseba category was to avoid polluting the higher-level one with infrequently-used programs. But it is fine to remove it.

**Comment by brettle** [comment]
Ok.  Changes made and builds succeeded.

**Comment by stephanemagnenat** [comment]
Thank you very much!
